ibm/service/cos/data_source_ibm_cos_bucket.go

```diff
@@ -327,14 +331,25 @@ func dataSourceIBMCosBucketRead(d *schema.ResourceData, meta interface{}) error
 	serviceID := d.Get("resource_instance_id").(string)
 	bucketType := d.Get("bucket_type").(string)
 	bucketRegion := d.Get("bucket_region").(string)
-	var endpointType = d.Get("endpoint_type").(string)
+	endpointType := d.Get("endpoint_type").(string)
+	stlocation := d.Get("satellite_location_id").(string)
+
+	if bucketType == "satellite_location_id" {
```

Comment:
bucket_type user will give has "satellite_location_id"? I think this is wrong if user provides "satellite_location_id" then you need to compute  bucket type as sl